docs/src/concepts/architectures.md

```diff
@@ -0,0 +1,43 @@
+# Architectures
+
+The abstract type `Architecture` can be concretized to be either an architecture of a concrete type `SingleDeviceArchitecture` or a `DistributedArchitecture` for MPI usage.
+
+## Single Device Architecture
+
+An object being the type of `SingleDeviceArchitecture` meaning that it contains the handle to a single CPU or GPU device with a respetive backend, where a device can be specified by an integer ID.
+
+### Backend Selection & Architecture Initialization
+
+We currently support CPU architectures, as well as CUDA and ROC backends for Nvidia and AMD GPUs through a thin wrapper around the [`KernelAbstractions.jl`](https://github.com/JuliaGPU/KernelAbstractions.jl) for users to select desirable backends.
+
+```julia
+# Default with CPU
+arch = Arch(CPU())
+```
+
+```julia
+# using CUDA
+arch = Arch(CUDABackend())
+```
+
+```julia
+# using AMDGPU
+arch = Arch(ROCBackend())
+```
+
+At the beginning of program, one may specify the backend and initialize the architecture they desire to use. The initialized `arch` variable will be required explicitly at creation of some entiries such as grids and kernel launchers.
+
+### Specifying Stream Priority
+
+For advanced users, we provide a convenient wrapper `activate!(arch::SingleDeviceArchitecture; priority=:normal)` for specifying the stream priority owned by the task one is executing. The stream priority will be set to `:normal` by default, where `:low` and `:high` are also possible options given that the target backend has priority control over streams implemented.
+
+For an example implementation, see [`AMDGPU.priority!`](https://amdgpu.juliagpu.org/stable/streams/#AMDGPU.priority!).
```

Comment:
The KA and thus backend-agnostic way to set priority is https://github.com/JuliaGPU/KernelAbstractions.jl/blob/b1d557bec65a7aaabf4bd00c6c5cb39821340912/src/KernelAbstractions.jl#L536-L552

Reply:
I see! But it seems like users of `Chmy.jl` would still be using just `activate!`, if they need to specify priority at all? Or how should I better edit this section here?

Reply:
I get your point. IDK what's better here, the thing is that KA priority is backend-agnostic and would de facto dispatch on the underlying GPU package priority setting. To be completes, one could say that KA implements the priority feature which further dispatches on specific backend, e.g., AMDGPU.jl and CUDA.jl.

Reply:
Good point, reformulated.